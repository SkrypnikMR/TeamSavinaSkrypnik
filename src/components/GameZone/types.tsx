import { ITheme } from '../App/types';

export interface IDiv extends ITheme{
    background: string;
    color: string;
    margin: string;
    textAlign: string;
    padding: string;
    maxHeight: string;
}

export type TGZ = {
    gameType: string;
}
