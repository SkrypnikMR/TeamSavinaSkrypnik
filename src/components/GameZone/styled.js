import styled from 'styled-components';

export const StGameZone = styled.div`
    background: #404040;
    color: white;
    margin: 10px 0;
    text-align: end;
    padding: 10px;
    max-height: 100%;
`;