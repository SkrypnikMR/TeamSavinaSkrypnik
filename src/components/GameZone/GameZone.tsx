import React from 'react';
import Chessplate from '../ChessPlate';
import TicTacToePlate from '../TicTacToePlate';
import { useTheme } from '../Hook/useTheme';
import { StGameZone } from './styled';
import { TGZ } from './types';

const GameZone = ({ gameType }: TGZ) => {
    const { colors, theme } = useTheme();
    switch (gameType) {
        case 'Checkers': return <StGameZone colors={colors} theme={theme}><Chessplate /></StGameZone>;
        case 'Tic-tac-toe': return <StGameZone colors={colors} theme={theme}><TicTacToePlate /></StGameZone>;
        default: return <StGameZone colors={colors} theme={theme}>Тут будет компонент приветствия</StGameZone>;
    }
};

export default GameZone;
