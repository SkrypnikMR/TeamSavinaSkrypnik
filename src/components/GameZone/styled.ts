import styled from 'styled-components';
import { white } from '../UI/baseLayout';

import { IDiv } from './types';

export const StGameZone = styled.div < IDiv > `
    background: ${({ colors, theme }) => colors[theme].background};
    color: ${white};
    text-align: end;
    padding: 10px;
    max-height: 100%;
    border-radius: 3px;
`;
