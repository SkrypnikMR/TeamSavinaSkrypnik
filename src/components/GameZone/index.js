import GameZone from "./GameZone";

export default GameZone