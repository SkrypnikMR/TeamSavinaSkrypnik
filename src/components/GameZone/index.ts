import { connect } from 'react-redux';
import GameZone from './GameZone';
import { getActualRoomGameType } from '../../store/game/selectors';

const mapStateToProps = store => ({ gameType: getActualRoomGameType(store) });

export default connect(mapStateToProps)(GameZone);
