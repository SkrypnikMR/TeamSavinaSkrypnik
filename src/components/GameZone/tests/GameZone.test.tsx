import React from 'react';
import { shallow, mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { act } from 'react-dom/test-utils';
import GameZone from '../GameZone';
import { mountSmart } from '../../../../__tests__/testHelper';

const mockStore = configureStore();
const store = mockStore({
  game: {
     rooms: [],
    userLogin: localStorage.getItem('login') || '',
    actualRoom: {
      gameType: '',
      creatorLogin: '',
      guestLogin: '',
      startTime: 0,
      id: '',
      stepDoList: [],
    },
    stepOrder: '',
    stepHistory: [],
    winner: '',
    possibleSteps: [],
  },
});

describe('GameZone', () => {
    let portalFolder: any;
    beforeEach(() => {
        portalFolder = document.createElement('div');
        portalFolder.setAttribute('id', 'portal');
        document.body.append(portalFolder);
    });
    afterEach(() => {
        document.body.removeChild(portalFolder);
    });
    const props = {
        gameType: '',
    };
    it('Should match snapshot', () => {
        const component = shallow(<GameZone {...props}/>);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render Hello', () => {
        const component = mount(<GameZone {...props}/>);
        expect(component.find('Hello')).toHaveLength(1);
    });
    it('should render case Checkers', () => {
        props.gameType = 'Checkers';
        const component = mountSmart(<GameZone {...props} />, store);
        expect(component.find('styled__StGameZone')).toHaveLength(1);
        expect(component.find('Turn')).toHaveLength(1);
        expect(component.find('Button')).toHaveLength(1);
        expect(component.find('Chessplate')).toHaveLength(1);
    });
    it('should render delete Modal', () => {
        props.gameType = 'Checkers';
        const component = mountSmart(<GameZone {...props} />, store);
            act(() => component.find('Button').props().onClick());
            component.update();
            expect(component.find('ModalCustom')).toHaveLength(1);
    });
        it('should render case Tic-Tac-Toe', () => {
        props.gameType = 'Tic-tac-toe';
        const component = mountSmart(<GameZone {...props} />, store);
        expect(component.find('styled__StGameZone')).toHaveLength(1);
        expect(component.find('Turn')).toHaveLength(1);
        expect(component.find('TicTacToePlate')).toHaveLength(1);
    });
});
