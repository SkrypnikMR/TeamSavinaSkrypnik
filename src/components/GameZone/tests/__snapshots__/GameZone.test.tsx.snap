// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameZone Should match snapshot 1`] = `"<div class=\\"styled__StHello-sc-3v9sfb-0 bwTzmh\\"><img src=\\"https://i.gifer.com/origin/51/518532f622d962be53e2e1f8989263a8_w200.gif\\"/><p>hi, Unknown!!!<br/>joing_room</p></div>"`;
